### PR TITLE
HPCC-31202	File permission dialog fails to open

### PR DESCRIPTION
**Pull request by kunalaswani**
When checking file permissions for file scopes, dialog was not opening. Dialog now opening for users to check permissions.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by kunalaswani** [comment]
@jeclrsg can you please review, i am having some trouble populating the user and group drop downs.

File: esp/src/src-react/components/forms/CheckPermissions.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                        onChange={onChange}
+                        label={nlsHPCC.Name}
+                        value={value}
+                        errorMessage={error && error?.message}
+                    />}
+                rules={{
+                    required: nlsHPCC.ValidationErrorRequired
+                }}
+            />
+            <Dropdown
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/CheckPermissions.tsx:98]: I don't think this "Type" dropdown is required. It's not in the dojo version of that dialog at least.

File: esp/src/src-react/components/forms/CheckPermissions.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    const { handleSubmit, control, reset, setValue, watch } = useForm<CheckPermissionsFormValues>({ defaultValues });
+
+    const [showError, setShowError] = React.useState(false);
+    const [errorMessage] = React.useState("");
+    const [userOptions, setUserOptions] = React.useState<IDropdownOption[]>([]);
+    const [groupOptions, setGroupOptions] = React.useState<IDropdownOption[]>([]);
+    const selectedType = watch("SelectedType");
+
+    React.useEffect(() => {
+        const fetchUsersAndGroups = async () => {
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/CheckPermissions.tsx:45]: Probably several errors in this, but I believe the inside of this useEffect here should be something like:
<pre>
React.useEffect(() => {
  WsAccess.CheckFilePermissions().then(({ FilePermissionResponse }) => {
    const { Users, Groups } = { ...FilePermissionResponse };

    setUserOptions(Users?.map(user => {
      return { key: user.username, text: user.username };
    }) ?? []);

    setGroupOptions(Groups?.map(group => {
      return { key: group.name, text: group.name };
    }) ?? []);
  }).catch(err => //etc...);
}, []);
</pre>

Where `WsAccess` would be imported from "src/WsAccess" (eg [.../components/forms/AddGroup.tsx#L6](https://github.com/hpcc-systems/HPCC-Platform/blob/master/esp/src/src-react/components/forms/AddGroup.tsx#L6))

**Review by jeclrsg — COMMENTED** [review]
@kunalaswani I think something like the comment about the React.useEffect where you currently have the `fetchUsersAndGroups` function will sort out the userOptions and groupOptions you mentioned

File: esp/src/src-react/components/forms/CheckPermissions.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+import { useForm, Controller } from "react-hook-form";
+import nlsHPCC from "src/nlsHPCC";
+import { MessageBox } from "../../layouts/MessageBox";
+import { FilePermission } from "src/ws_access";
+
+const logger = scopedLogger("src-react/components/forms/CheckPermissions.tsx");
+
+interface CheckPermissionsFormValues {
+    FileName: string;
+    SelectedValue: string;
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/CheckPermissions.tsx:13]: Should have separate values for user & group. Perhaps UserName & GroupName?

File: esp/src/src-react/components/forms/CheckPermissions.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        setShowForm(false);
+        setFilePermissionResponse("");
+    }, [reset, setShowForm]);
+
+    const onSubmit = async (data: CheckPermissionsFormValues) => {
+        try {
+            const { FileName, SelectedValue } = data;
+            logger.info(`Checking permissions for file ${FileName} for selected value: ${SelectedValue}`);
+            const response = await FilePermission({
+                request: {
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/CheckPermissions.tsx:75]: This is why SelectedValue should be separated into two. The FilePermission response is not reliable if both UserName and GroupName are submitted with the same value.

File: esp/src/src-react/components/forms/CheckPermissions.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+                rules={{
+                    required: nlsHPCC.ValidationErrorRequired
+                }}
+            />
+            <Dropdown
+                label={nlsHPCC.Users}
+                options={userOptions}
+                selectedKey={watch("SelectedValue")}
+                onChange={(ev, option) => {
+                    setValue("SelectedValue", option?.key.toString() || "");
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/CheckPermissions.tsx:116]: Similarly, the dojo version will blank out the other field when one changes, ie selecting a User will unset the selected Group if one was selected.

File: esp/src/src-react/components/forms/CheckPermissions.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                }}
+            />
+            <Dropdown
+                label={nlsHPCC.Groups}
+                options={groupOptions}
+                selectedKey={watch("SelectedValue")}
+                onChange={(ev, option) => {
+                    setValue("SelectedValue", option?.key.toString() || "");
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/CheckPermissions.tsx:124]: And vice versa

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@kunalaswani Noticed a couple small things that should be changed. Really just one thing, but it's in a couple or three different places

File: esp/src/src-react/components/forms/CheckPermissions.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                request: {
+                    FileName,
+                    UserName,
+                    GroupName
+                }
+            });
+            setFilePermissionResponse(response.FilePermissionResponse.UserPermission);
+        } catch (error) {
+            logger.error(error);
+            setErrorMessage("Error occurred while fetching file permissions.");
```
Comment (jeclrsg) [inline esp/src/src-react/components/forms/CheckPermissions.tsx:86]: Should this error message be added to the translation file?

**Review by jeclrsg — APPROVED** [review]
@kunalaswani This looks like it would be checking the file permissions correctly. Had a question about whether that error message should be added to the translations file? But I'd approve otherwise

**Comment by ghalliday** [comment]
@kunalaswani seem comment above about translation

File: esp/src/src-react/components/forms/CheckPermissions.tsx (hunk 7 above)
Comment (kunalaswani) [reply esp/src/src-react/components/forms/CheckPermissions.tsx:86]: Yes i think it should, i will add it to the file.

**Comment by kunalaswani** [comment]
@ghalliday error message added to translations.

**Pull request merged**